extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/OidcBuildTimeConfig.java

```diff
@@ -13,4 +14,10 @@
      */
     @ConfigItem(defaultValue = "true")
     public boolean enabled;
+
+    /**
+     * Dev services configuration.
+     */
+    @ConfigItem
+    public DevServicesConfig devservices;
```

Comment:
This config is keycloak specific, shouldn't it be under quarkus.keycloak rather than oidc?

Reply:
Hi Stuart, @stuartwdouglas 
Yes, I was thinking about it, Keycloak is a `default` devservices option, hence I thought about avoiding `keycloak`. The docs will say that `quarkus.oidc` will launch a KC container by default in the dev mode but it can be disabled with `quarkus.oidc.devservices.enabled=false`

I'm also looking at how to handle the case where the OidcDebugger only Dev UI is presented (or if a Quarkiverse `oidc-okta` etc is available), i.e, if I'm an Okta user then I feel I'd rather do:
```
# disable the default devservices option:
quarkus.oidc.devservices.enabled=false
# Configure Okta container, etc
quarkus.okta.devservices....
``` 

as opposed to
```
# disable the default devservices option:
quarkus.keycloak.devservices.enabled=false
# Configure Okta container, etc
quarkus.okta.devservices....
```

in this last case a `keycloak` property looks a bit intrusive to me as an OKta user :-) - I'd like to avoid introducing a requirement for non-KC users to refer to Keycloak in their configuartion.

I think that if we, at a later point, either still in this PR of after it has been merged, decide to return to the idea of shipping a dedicated `quarkus-oidc-keycloak` then indeed it would be `quarkus.keycloak.devservices...`.

The alternative is indeed to have `quarkus.keycloak.devservices` namespace within `quarkus-oidc` but disable `quarkus.keycloak.devservices.enabled=false` by default - in this case the only users who will be dealing with `quarkus.keycloak.devservices` will be Keycloak users.

I'm getting a bit dizzy with all these options :-), perhaps I'm over-thinking it. But I believe it is important to keep a 'neutral` aspect of `quarkus-oidc` when it comes to configuring it as well.

So I'd either 1) keep it as is now (since KC is default for devservices, `quarkus.oidc.devservices` namespace if used to configure it) or we do 2) `quarkus.keycloak.devservices` within `quarkus-oidc` but disable by default

Perhaps 2) is not bad - recall, OOB, there will eventually be an `OidcDebugger` like Dev UI experience anyway where one would expect a container started separately or an Auth0/etc account is available, etc. To get a KC devservices experience KC users would have to do `quarkus.keycloak.devservices.enabled=true` - extra effort - but it is comparable to effort which would be required if they had to choose `quarkus-oidc-keycloak` (instead of `quarkus-oidc`) to get this experience.

IMHO, perhaps staying with 1) is OK for now but we should review 2) if you prefer to keep it all under the `quarkus.keycloak.devservices` namespace

let me know what you prefer please, thanks

Reply:
@stuartwdouglas I have tried `oidc/deployment` tests with this PR and I've found a few issues related to `quarkus.oidc.devservices.enabled=true` (which is a default):
1. Dev Mode test which starts without `quarkus.oidc.auth-server-url` and verifies the auth request fails actually passes (or rather fails to fail :-) ) since the dev services container is started and this property is provisioned - so I had to disable the devservices to verify it passes
2. if we have a case where OIDC multi-tenancy is tested (without the default config, without `quarkus.oidc.auth-server-url`, which is usually the case when `TenantConfigResolver` is used to create the config dynamically at the request time) then the devservices container will also be starting during the tests
3. Normal non-dev mode security disabled test (which is also run by QE) also spawns a container - test passes - but slowly

For 1. and 3. I could fix the problem by disabling the devservices in the config - for 2. - it is not really fixable - or rather, I don't know if starting the devservices container can be prevented.

All of these issues would stay even if we revert to the idea of having `quarkus-oidc-keycloak` and disabling the devservices there would be worse since it would be an extension designed to offer KC devservices exp out of the box.

Given this and the earlier comment, I'm coming to the conclusion that the best option is to have `quarkus.keycloak.devservices.enabled=false` by default (with the configuration being in the `quarkus.keycloak.devservices` namespace as you suggested).

So when a user will start `quarkus-oidc` with the empty `application.properties` - the empty OIDC card will be shown. Next user goes to application.properties and adds `quarkus.keycloak.devservices.enabled=false` - and the container has started - or the user adds `auth-server-url` - in this case OIDC card will show a link to an OIDC page where one would do an OidcDebugger like view. `quarkus-okta`/`quarkus-auth0` benefit as well - there would be no need to disable the default devservices. I'll prototype it shortly, IMHO this works for all the cases

Reply:
I think it should be quarkus.keycloak.devservices, if we have a quarkus-okta then we can use build items to stop keycloak being enabled by default (i.e. quarkus-okta would provide a build item that would disable the keycloak devservices).

Basically this means that keycloak is the default and works out of the box, but if you add another extension it turns off keycloak. 

In terms of your test problems:

1) is completly fine, other tests have had this problem, just turn it off for the specific test
2) I think if multi tencancy is enabled the devservices should probably be disabled. Not 100% sure about this though, maybe just turn it off for the test
3) If security is disabled why isn't devservices disabled?

The whole point of dev services is that it is zero config, it should not have to be specifically enabled.

Reply:
Hi Stuart @stuartwdouglas, it all makes sense, I've already changed the namespace yesterday, and good idea about disabling KC devservices indirectly via the dedicated build item (this item can also provide some extra info). OK, tests can be taken care of as well.

> The whole point of dev services is that it is zero config, it should not have to be specifically enabled.

Right this is where the design `tension` is :-), yes, I totally agree we'd like the KC users to have the lowest effort applied toward enabling the dev services; but what somehow concerns me is that if you are not a KC user you won't have a zero effort at starting `quarkus:dev` - `quarkus-okta`/`quarkus-auth0` are just the ideas yet to be realized by those provider's experts which may or may not happen - so these users would have to type `quarkus.keycloak.devservices.enabled=false` before starting `quarkus:dev` - they don't have to do it now - while after this PR they will have to and a `keycloak` config root would become a permanent feature of their configuration - `quarkus-oidc` will lose some of its `neutrality` toward offering a provider neutral experience. This is the reason behind my hesitation to enable it OOB.

Let me do the final round of various minor updates, docs, and I'll postpone making `quarkus.keycloak.devservices.enabled=true` a default value till the last moment and then I'll ping you to finalize :-) It is nearly there :-)

Reply:
If you are not using KC then you will have to configure a URL, same as what you do now, and this will disable DevServices.

If we have support for okta/auth0 then you would just include these extensions and then DevServices would stop starting KC and start the other one instead.